File: CmdMents/TimesAndSeasons/HearTorahAtEndOf7YearsDuringSukkot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CmdMents.TimesAndSeasons
{
    class HearTorahAtEndOf7YearsDuringSukkot : LiveInSukkahDuringSukkot
    {
        public HearTorahAtEndOf7YearsDuringSukkot()
        {
            this.AlternateText = null;
            this.Book = CommandmentBook.Deuteronomy;
            this.CanBeCarriedOutOnlyInIsrael = false;
            this.CanBeCarriedOutToday = true;
            this.Chapter = 32;
            this.CommandmentType = CmdMents.CommandmentType.Positive;
            this.FollowedByChristians = CommandmentObedience.None;
            this.FollowedByMessianics = CommandmentObedience.None;
            this.FollowedByObservantJews = CommandmentObedience.None;
            this.Number = 425;
            this.ShortSummary = "Assemble the people to hear the Torah during the 7th year on Sukkot.";
            this.Text = "Then Moses commanded them: “At the end of every seven years, in the year for canceling debts, during the Festival of Tabernacles, when all Israel comes to appear before the LORD your God at the place he will choose, you shall read this law before them in their hearing. Assemble the people—men, women and children, and the foreigners residing in your towns—so they can listen and learn to fear the LORD your God and follow carefully all the words of this law.";
            this.Verse = 12;
            this.Meta(CommandmentMetaType.Commentary, "Interesting commandment indeed: when all Israel goes up to the Tabernacle for the pilgrim feasts (Unleavened Bread, Shavuot, Sukkot), on the 7th year, which apparently is a kind of Jubilee year, the Torah is to be read to all the people.");
            this.Meta(CommandmentMetaType.MaimonidesInterpretation, "To assemble all the people on the Sukkot following the seventh year.");
        }
    }
}

File: EtzMitzvot/Commandments/TimesAndSeasons/hearTorahAtEndOf7YearsDuringSukkot.ts
///<reference path='../commandmentBase.ts' />

class HearTorahAtEndOf7YearsDuringSukkot extends CommandmentBase {
    constructor() {
        super();

        this.shortSummary = "Hear the Torah read in Jerusalem <br />on Sukkot following the 7th year.";
        this.text = "Then Moses commanded them: “At the end of every seven years, in the year for canceling debts, during the Festival of Sukkot, when all Israel comes to appear before the LORD your God at the place he will choose, you shall read this law before them in their hearing. Assemble the people—men, women and children, and the foreigners residing in your towns—so they can listen and learn to fear the LORD your God and follow carefully all the words of this law.";
        this.book = TorahBook.Deuteronomy;
        this.chapter = 32;
        this.verse = 12;

        this.canBeCarriedOutToday = true;
        this.requiresLivingInIsrael = false;
        this.requiresTemple = false;
        this.commentary = "The text explicitly calls for hearing the Torah in the place that God chooses (Jerusalem). However, this doesn't require living in Israel, as those outside Jerusalem (or even outside Israel) would make pilgrimage. Therefore, I've marked this commandment as not requiring living in Israel. Additionally, I've labeled this commandment as able to be kept today, even though the text calls for a reading of the Torah on the 'place God Chooses', which, if this is the Temple Mount, it would be impossible to carry out this commandment due to Islamic restrictions for non-Muslims on the Temple Mount.";

        this.type = CommandmentType.Positive;
        this.christianObservance = CommandmentObedience.None;
        this.messianicObservance = CommandmentObedience.None;
        this.jewishObservance = CommandmentObedience.RecognizedButPrevented; // I need to verify this. Since this is "the place God chooses" (the Temple mount), maybe this can't be carried out.
        this.number = 425;

        this.children = [

        ];
    }
}  